packages/fx-core/src/plugins/resource/bot/v2/bicep.ts

```diff
@@ -0,0 +1,162 @@
+      (p) => new NamedArmResourcePluginAdaptor(p)
+    );
+    const pluginCtx = {
+      plugins: plugins.map((obj) => obj.name),
+      configurations: configuration,
+    };
+    const bicepTemplateDir = path.join(
+      getTemplatesFolder(),
+      BicepTemplateRelativeDir,
+      FunctionBicepFolderName
+    );
+    const bicepFilenames = [
+      Bicep.ProvisionFileName,
+      Bicep.ConfigFileName,
+      "functionProvision.template.bicep",
+      "functionConfiguration.template.bicep",
+    ];
+    let modules = await Promise.all(
+      bicepFilenames.map((name) =>
+        generateBicepFromFile(path.join(bicepTemplateDir, name), pluginCtx)
+      )
+    );
+    modules = modules.map((module) => module.replace(/PluginIdPlaceholder/g, "fx-resource-bot"));
+    const result: ArmTemplateResult = {
+      Provision: {
+        Orchestration: modules[0],
+        Modules: { function: modules[2] },
+      },
+      Configuration: {
+        Orchestration: modules[1],
+        Modules: { function: modules[3] },
+      },
+      Reference: {
+        resourceId: functionResourceId,
+        hostName: functionHostName,
+        functionEndpoint: functionEndpoint,
+        endpointAsParam: endpointAsParam,
+      },
+    };
+    return result;
+  }
+
+  async updateBicep(ctx: Context, configuration: Configurations): Promise<ResourceTemplate> {
+    return {} as ArmTemplateResult;
+  }
+}
+
+export class BotServiceBicepGenerator implements BicepGenerator {
+  async generateBicep(ctx: Context, configuration: Configurations): Promise<ResourceTemplate> {
+    const plugins = getActivatedV2ResourcePlugins(ctx.projectSetting).map(
+      (p) => new NamedArmResourcePluginAdaptor(p)
+    );
+
+    const pluginCtx = { plugins: plugins.map((obj) => obj.name) };
+    const bicepTemplateDir = path.join(
+      getTemplatesFolder(),
+      BicepTemplateRelativeDir,
+      BotServiceBicepFolderName
+    );
+    const bicepFilenames = [Bicep.ProvisionFileName, "botserviceProvision.template.bicep"];
+
+    const modules = await Promise.all(
+      bicepFilenames.map((name) =>
+        generateBicepFromFile(path.join(bicepTemplateDir, name), pluginCtx)
+      )
+    );
+    const result: ArmTemplateResult = {
+      Provision: {
+        Orchestration: modules[0],
+        Modules: { bot: modules[1] },
+      },
+      Parameters: await fs.readJson(path.join(bicepTemplateDir, Bicep.ParameterFileName)),
+    };
+    return result;
+  }
+
+  async updateBicep(ctx: Context, configuration: Configurations): Promise<ResourceTemplate> {
+    return {} as ArmTemplateResult;
+  }
+}
+
+export function getGenerators(ctx: Context, inputs: Inputs): BicepGenerator[] {
+  const generators = [];
+  const rawHostType = ctx.projectSetting?.pluginSettings?.[PluginBot.PLUGIN_NAME]?.[
+    PluginBot.HOST_TYPE
+  ] as string;
+
+  const hostType = utils.convertToConstValues(rawHostType, HostTypes);
+
+  if (hostType === BotHostTypes.AppService) {
```

Comment:
Is it possible to encapsulate the creation logic to separate Factory and its input is hostType, including function, appservice, azurebot.
In this way, Bot plugin only need to declare azurebot and hosting generator from Factory.